vllm/model_executor/guided_decoding/__init__.py

```diff
@@ -122,25 +132,34 @@ async def get_guided_decoding_logits_processor(
         from vllm.model_executor.guided_decoding.xgrammar_decoding import (  # noqa
             get_local_xgrammar_guided_decoding_logits_processor)
         return get_local_xgrammar_guided_decoding_logits_processor(
-            guided_params, tokenizer, model_config)
+            guided_params, tokenizer, model_config, reasoner_config)
 
     raise ValueError(
         f"Unknown guided decoding backend '{guided_params.backend}'. "
         "Must be one of 'outlines, 'lm-format-enforcer', 'xgrammar'")
 
 
 def get_local_guided_decoding_logits_processor(
-        guided_params: GuidedDecodingParams, tokenizer: PreTrainedTokenizer,
-        model_config: ModelConfig) -> LogitsProcessor | None:
+        guided_params: GuidedDecodingParams,
+        tokenizer: PreTrainedTokenizer,
+        model_config: ModelConfig,
+        reasoning_backend: str | None = None) -> LogitsProcessor | None:
     guided_params = maybe_backend_fallback(guided_params)
+
+    # Get the reasoner config if a reasoning backend is specified
+    reasoner_config = None
+    if reasoning_backend is not None:
+        reasoner = get_reasoner(reasoning_backend, tokenizer)
+        reasoner_config = ReasonerConfig.from_reasoner(reasoner)
+
     # CFG grammar not supported by LMFE, so we use outlines instead
     if guided_params.backend_name == 'outlines':
         # NOTE: lazy import outlines to avoid https://github.com/vllm-project/vllm/issues/4193
         from vllm.model_executor.guided_decoding.outlines_decoding import (  # noqa
             get_local_outlines_guided_decoding_logits_processor)
         return get_local_outlines_guided_decoding_logits_processor(
-            guided_params, tokenizer)
-    if guided_params.backend_name == 'lm-format-enforcer':
+            guided_params, tokenizer, reasoner_config)
+    if guided_params.backend == 'lm-format-enforcer':
```

Comment:
```suggestion
    if guided_params.backend_name == 'lm-format-enforcer':
```

for a recent change on main for `xgrammar:no-fallback`

Reply:
@gaocegege

Reply:
I am not sure why I rebased the upstream main branch but it is not changed. Will update it.